src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs

```diff
@@ -515,6 +516,22 @@ public void Dispose()
         this.scanDecoder = null;
     }
 
+    /// <summary>
+    /// Assigns COM marker bytes to comment property
+    /// </summary>
+    /// <param name="stream">The input stream.</param>
+    /// <param name="markerContentByteSize">The remaining bytes in the segment block.</param>
+    private void ProcessComMarker(BufferedReadStream stream, int markerContentByteSize)
+    {
+        Span<byte> temp = new byte[markerContentByteSize];
+        JpegMetadata metadata = this.Metadata.GetFormatMetadata(JpegFormat.Instance);
+
+        stream.Read(temp);
+        string comment = Encoding.ASCII.GetString(temp);
```

Comment:
We're assuming here that the input is always ASCII when we do not actually know that. I would create a `char[]` cast it to `byte[]`, read the stream and pass it to the `JpegComData` constructor.